cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/ProgramRunInfo.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.runtimejob;
+
+import java.util.Objects;
+
+/**
+ * Program run information.
+ */
+public class ProgramRunInfo {
+  private final String namespace;
+  private final String application;
+  private final String program;
+  private final String run;
+
+  public ProgramRunInfo(String namespace, String application, String program, String run) {
```

Comment:
You also need programType?

Reply:
From the extension perspective type is not needed. But we can add it.